src/services/findAllReferences.ts

```diff
@@ -1425,6 +1429,19 @@ namespace ts.FindAllReferences {
             }
         }
 
+        function getAllReferencesForImportMeta(sourceFiles: readonly SourceFile[], cancellationToken: CancellationToken): SymbolAndEntries[] | undefined {
+            const references = flatMap(sourceFiles, sourceFile => {
+                cancellationToken.throwIfCancellationRequested();
+                return mapDefined(getPossibleSymbolReferenceNodes(sourceFile, "meta", sourceFile), node => {
+                    const parent = node.parent;
+                    if (isImportMeta(parent)) {
+                        return nodeEntry(parent);
+                    }
+                });
+            });
+            return references.length ? [{ definition: { type: DefinitionKind.Keyword, node: references[0].node }, references }] : undefined;

```

Comment:
I don't exactly know if you should be considering anything a "definition" location apart from the global `ImportMeta` interface.

Even that one is a little weird, because that's what you'd want if you requested "go-to-type". Do all responses require a `definition`? @jessetrinity and @andrewbranch might have more knowledge here.

Reply:
The types imply that a definition isn’t required. My intuition is that this shouldn’t be one.

Reply:
It's copied from `getAllReferencesForKeyword`, which searches primitive types like `string/number/any`. They should have no definition either?

```ts
function getAllReferencesForKeyword(sourceFiles: readonly SourceFile[], keywordKind: SyntaxKind, cancellationToken: CancellationToken, filter?: (node: Node) => boolean): SymbolAndEntries[] | undefined {
    // ...
    return references.length ? [{ definition: { type: DefinitionKind.Keyword, node: references[0].node }, references }] : undefined;
}
```

Reply:
I’m not 100% sure what the implications are for removing it—we should check with @mjbvz ad @uniqueiniquity.

Reply:
Let’s get rid of the `definition`

Reply:
`findReferencedSymbols` won't work if `definition` is absent.
https://github.com/microsoft/TypeScript/blob/29c7ae23cadd3ef866ebddc3923423eec27fc531/src/services/findAllReferences.ts#L207-L219

Reply:
🤦 I swear all this stuff was just made up on the spot